lib/wifi_user/use_case/parse_email_request.rb

```diff
@@ -9,16 +9,36 @@ def execute(request)
     logger.debug("Processing request: #{parsed_request} with message #{parsed_message}")
 
     {
-      type: parsed_request['Type'],
-      message_id: parsed_message['mail']['messageId'],
-      from_address: parsed_message['mail']['commonHeaders']['from'][0],
-      to_address: parsed_message['mail']['commonHeaders']['to'][0],
-      s3_object_key: parsed_message['receipt']['action']['objectKey'],
-      s3_bucket_name: parsed_message['receipt']['action']['bucketName']
+      type: parsed_request.fetch('Type'),
+      message_id: message_id(parsed_message),
+      from_address: from_address(parsed_message),
+      to_address: to_address(parsed_message),
+      s3_object_key: s3_object_key(parsed_message),
+      s3_bucket_name: s3_bucket_name(parsed_message)
     }
   end
 
 private
 
   attr_reader :logger
+
+  def message_id(request)
```

Comment:
Interestingly we just took the opposite approach in another PR - in-lining a bunch of one-line methods.

I think this is the right choice here, though. Hide some of this nested fetch horribleness.